### PR TITLE
Added reusable components and created documentation

### PR DESCRIPTION
**Pull request by tims139**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -72,6 +73,7 @@ function App() {
             <Route path="/about" element={<About />} />
             <Route path="/contact-us" element={<ContactUs />} />
             <Route path="/login" element={<LogIn />} />
+            <Route path="/documentation" element={<Documentation />} />
```
Comment (akhilwaghmare) [inline src/App.jsx:77]: Can you add a comment to remind us to comment this route out before deploying?

File: src/components/BaseLink.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+function BaseLink({ to, children, className }) {
+  return (
+    <Link to={to} className={`text-blue-500 hover:text-blue-700 ${className}`}>
```
Comment (akhilwaghmare) [inline src/components/BaseLink.jsx]: Suggestion - add a transition for the hover effect. You can leverage tailwind classes like `transition-colors` and `duration-200`

File: src/components/Button.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+function Button({ children, className, ...props }) {
+  return (
+    <button
```
Comment (akhilwaghmare) [inline src/components/Button.jsx:5]: Similar to `BaseLink`, think about adding an animation for the hover effect

**Review by akhilwaghmare — APPROVED** [review]
Nice job on this. I left some suggestions for adding a couple animations for hover effects, but not required.

File: src/App.jsx (hunk 1 above)
Comment (tims139) [reply src/App.jsx:77]: Done!

File: src/components/BaseLink.jsx (hunk 2 above)
Comment (tims139) [reply src/components/BaseLink.jsx]: Done!

File: src/components/Button.jsx (hunk 3 above)
Comment (tims139) [reply src/components/Button.jsx:5]: Done!

**Pull request merged**